python/beeai_framework/agents/tool_calling/agent.py

```diff
@@ -65,13 +65,30 @@ def __init__(
         self._templates = to_model(ToolCallingAgentTemplates, templates or {})
 
     async def _run(
-        self, run_input: ToolCallingAgentRunInput, options: ToolCallingAgentRunOptions, context: RunContext
+        self,
+        run_input: ToolCallingAgentRunInput,
+        options: ToolCallingAgentRunOptions,
+        context: RunContext,
     ) -> ToolCallingAgentRunOutput:
         memory = UnconstrainedMemory()
         await memory.add(SystemMessage(self._templates.system.render()))
         await memory.add_many(self.memory.messages)
+
         if run_input.prompt is not None:
-            await memory.add(UserMessage(run_input.prompt))
+            if (
+                run_input.expected_output is not None
+                and isinstance(run_input.expected_output, type)
+                and issubclass(run_input.expected_output, BaseModel)
+            ):
+                await memory.add(
+                    UserMessage(
+                        self._templates.task.render(
+                            ToolCallingAgentRunInput(**run_input.model_dump(exclude={"expected_output"}))
```

Comment:
I think we might consider adding the expected output to the schema. This can prevent us from doing additional iteration.